src/Persistence/Array_/ArrayByRef.php

```diff
@@ -0,0 +1,109 @@
+<?php
```

Comment:
Maybe this class could be useful somewhere else too and if so then we can move it to atk4/core ?

Reply:
I spend some time with it and I would advise to not to do it now - the new class can never replace arrays at all - for example set by reference is not possible (due to the ArrayAccess iface) and also there is not good type for it - the best is iterable, array accepts only arrays...